packages/model-viewer/src/test/features/scene-graph/mesh-primitive-spec.ts

```diff
@@ -0,0 +1,134 @@
+};
+
+suite('scene-graph/model/mesh-primitives', () => {
+  suite('Static Primitive', () => {
+    test('Primitive count matches glTF file', async () => {
+      const threeGLTF = await loadThreeGLTF(SHEEN_CHAIR_GLB_PATH);
+      const model = new Model(CorrelatedSceneGraph.from(threeGLTF));
+      expect(model![$primitives].length).to.equal(4);
+    });
+
+    test('Should have variant info', async () => {
+      const threeGLTF = await loadThreeGLTF(SHEEN_CHAIR_GLB_PATH);
+      const model = new Model(CorrelatedSceneGraph.from(threeGLTF));
+      expect(findPrimitivesWithVariant(model, 'Mango Velvet')).to.not.be.null;
+      expect(findPrimitivesWithVariant(model, 'Peacock Velvet')).to.not.be.null;
+    });
+
+    test('Should not have variant info', async () => {
+      const threeGLTF = await loadThreeGLTF(SHEEN_CHAIR_GLB_PATH);
+      const model = new Model(CorrelatedSceneGraph.from(threeGLTF));
+
+      let hasNoVariantInfoCount = 0;
+      for (const primitive of model![$primitives]) {
+        if (primitive.variantInfo === undefined) {
+          hasNoVariantInfoCount++;
+        }
+      }
+      expect(hasNoVariantInfoCount).equals(2);
+    });
+
+    test('Switching to incorrect variant name', async () => {
+      const threeGLTF = await loadThreeGLTF(SHEEN_CHAIR_GLB_PATH);
+      const model = new Model(CorrelatedSceneGraph.from(threeGLTF));
+      const primitive = findPrimitivesWithVariant(model, 'Mango Velvet')![0];
+      const material = await primitive.enableVariant('Does not exist');
+      expect(material).to.be.null;
+    });
+
+    test('Switching to current variant', async () => {
+      const threeGLTF = await loadThreeGLTF(SHEEN_CHAIR_GLB_PATH);
+      const model = new Model(CorrelatedSceneGraph.from(threeGLTF));
+      const primitives = findPrimitivesWithVariant(model, 'Mango Velvet')!;
+      const materials = new Array<MeshStandardMaterial>();
+      for (const primitive of primitives) {
+        materials.push(
+            primitive.enableVariant('Mango Velvet') as MeshStandardMaterial);
+      }
+      expect(materials).to.not.be.empty;
+      expect(materials.find((material: MeshStandardMaterial) => {
+        return material.name === 'fabric Mystere Mango Velvet';
+      })).to.not.be.null;
+    });
+
+    test('Switching to other variant name', async () => {
+      const threeGLTF = await loadThreeGLTF(SHEEN_CHAIR_GLB_PATH);
+      const model = new Model(CorrelatedSceneGraph.from(threeGLTF));
+      const primitives = findPrimitivesWithVariant(model, 'Mango Velvet')!;
+      const materials = new Array<MeshStandardMaterial>();
+      for (const primitive of primitives) {
+        materials.push(
+            primitive.enableVariant('Peacock Velvet') as MeshStandardMaterial);
+      }
+      expect(materials.find((material: MeshStandardMaterial) => {
+        return material.name === 'fabric Mystere Peacock Velvet';
+      })).to.not.be.null;
+    });
+
+    test('Switching back variant name', async () => {
```

Comment:
I might be tempted to combine the previous two tests into this one.

Reply:
Agreed